lua/lualine/highlight.lua

```diff
@@ -10,7 +10,41 @@ local function highlight (name, foreground, background, special)
   return table.concat(command, ' ')
 end
 
+local function apply_defaults_to_theme(theme)
+  local modes = {'insert', 'visual', 'replace', 'command', 'terminal', 'inactive'}
+  -- normal mode cann't have a fallback for now
+  local sections = {'a', 'b', 'c', 'x', 'y', 'z'}
+  for _, mode in ipairs(modes) do
+    repeat
+      if theme[mode] == nil then
+        theme[mode] = theme['normal']
+        break
+      end
+      for _, section in ipairs(sections) do
+        repeat
+          if theme[mode][section] == nil then
+            if section == 'x' then
+              theme[mode][section] = theme[mode]['c']
+            elseif section == 'y' then
+              theme[mode][section] = theme[mode]['b']
+            elseif section == 'z' then
+              theme[mode][section] = theme[mode]['a']
+            else
+              theme[mode][section] = theme['normal'][section]
```

Comment:
Will this yield the same results? I think it's a lot cleaner and still has room for improvements.
```lua
  for _, mode in ipairs(modes) do
    repeat
      theme[mode] = (theme[mode] or theme['normal'])
      theme[mode]['x'] = (theme[mode]['x'] or theme[mode]['c'])
      theme[mode]['y'] = (theme[mode]['y'] or theme[mode]['b'])
      theme[mode]['z'] = (theme[mode]['z'] or theme[mode]['a'])
      for _, section in ipairs(sections) do
        repeat
          theme[mode][section] = theme['normal'][section]
          break
        until true
      end
      break
    until true
  end
```

Reply:
nice . I like it . I'll update it .